src/OgContextInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Drupal\og;
+
+/**
+ * @file
+ * Contains \Drupal\og\OgContextInterface.
+ */
+
+use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\Core\Plugin\Context\ContextProviderInterface;
+
+/**
+ * Defines an interface for OG context plugins.
+ */
+interface OgContextInterface extends PluginInspectionInterface, ContextProviderInterface {
+
+  /**
+   * Return the current group. If no group will be found return Null.
+   *
+   * @return ContentEntityBase
+   *   Return the best matching group.
+   */
+  public function getGroup();
```

Comment:
Returning a single group will not be sufficient for some plugins. For example a plugin that matches a group content entity on the current route might need to return multiple plugins. Group content can reference multiple groups. The plugin cannot make a choice here which of the referenced groups is more important.